central/graphql/resolvers/loaders/nodes.go

```diff
@@ -58,16 +60,38 @@ type nodeLoaderImpl struct {
 
 // FromIDs loads a set of nodes from a set of ids.
 func (ndl *nodeLoaderImpl) FromIDs(ctx context.Context, ids []string) ([]*storage.Node, error) {
-	nodes, err := ndl.load(ctx, ids)
+	nodes, err := ndl.load(ctx, ids, false)
 	if err != nil {
 		return nil, err
 	}
 	return nodes, nil
 }
 
-// FromID loads a node from an ID.
+// FromID loads a node from an ID. Does not load scan components and vulnerabilities when Postgres is enabled.
 func (ndl *nodeLoaderImpl) FromID(ctx context.Context, id string) (*storage.Node, error) {
-	nodes, err := ndl.load(ctx, []string{id})
+	nodes, err := ndl.load(ctx, []string{id}, false)
+	if err != nil {
+		return nil, err
+	}
+	return nodes[0], nil
+}
+
+// FullNodeWithID loads full node from an ID.
+func (ndl *nodeLoaderImpl) FullNodeWithID(ctx context.Context, id string) (*storage.Node, error) {
+	node, err := ndl.FromID(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	// Load the full node if full scan is not available.
```

Comment:
The comment does not describe the if block below. A converse comment would work or move the comment after the if block.